Create findedit

halkoma opened the pull request:
1. Find files that contain '<search_term>' using rg
2. Use fzf with --preview to select the correct one
3. Use xargs -o to edit in neovim

halkoma committed:
Create findedit

1. Find files that contain '<search_term>' using rg
2. Use fzf with --preview to select the correct one
3. Use xargs -o to edit in neovim

jukuisma commented:
Love it! :joy_cat:

I'll have a closer look and merge when I'm off work.

p.s. publish your own dots so I can repay the kindness!

jukuisma commented:
In a similar vein, you might enjoy: https://github.com/jukuisma/dotfiles/blob/main/bin/syscalls.
![image](https://github.com/user-attachments/assets/ad6277b3-63d3-4c23-bdc1-1084e05a67a6)

halkoma commented:
> p.s. publish your own dots so I can repay the kindness!

Certainly, will look into it. Gotta tidy it up a little bit first to remove e.g. some old mutt stuff that contain some minor things I don't want to reveal. Not even using mutt anymore as it was a little bit PITA after all :smile:

halkoma commented:
> > p.s. publish your own dots so I can repay the kindness!
> 
> Certainly, will look into it. Gotta tidy it up a little bit first to remove e.g. some old mutt stuff that contain some minor things I don't want to reveal. Not even using mutt anymore as it was a little bit PITA after all 😄

Actually, did it right away, re-created the repo: https://github.com/halkoma/dotfiles/

jukuisma committed:
Minor changes to findedit script

1) `chmod +x bin/findedit` so this will work out of the box after `$ make bin`.
2) Remove the wrapping function so this works as a standalone script.
3) Check if user selected something and don't open neovim if they didn't.
4) Rename batcat to bat as that's what most distros ship and I have that
   symlinked even on debian: `.local/bin/bat -> /usr/bin/batcat*`.

jukuisma commented:
Pushed minor changes mostly specific to my environment here: https://github.com/jukuisma/dotfiles/pull/1/commits/d6c8e35f998377de3c9a16d13c2ea671bae68ff9. Anything to add, ship it?

halkoma commented:
Ah yes, missed the option of not opening nvim when no file was selected. It's better. I thought if it would be useful to do nothing if more than 50 files were found (like in `bin/replace`) but didn't find any reasons, really.

Only now noticed your Makefile way of doing things - just beautifully elegant ♥️

halkoma commented on bin/findedit:
```diff
@@ -0,0 +1,20 @@
+
+searchterm="$1"
+file=$(
+    rg -l "$searchterm" | \
```
```suggestion
    rg --vimgrep "$searchterm" | \
```

jukuisma replied:
vimgrep broke bat preview, but that and some other minor changes here: 214cc2269ef4e0b4a9e37f07d9a42353e07c1bd4. It's about time to merge this. :joy:

halkoma commented on bin/findedit:
```diff
@@ -0,0 +1,20 @@
+        --preview-window=bottom,80%
+)
+
+if [ -n "$file" ]; then
+    nvim "$file"
+fi
```
```suggestion
if [ -n "$selection" ]; then
  file=$(echo "$selection" | cut -d: -f1 )
  line=$(echo "$selection" | cut -d: -f2 )
  nvim +"$line | normal! zz" "$file"
fi
```

halkoma replied:
Now we have line numbers. Open file at the matched line and center with zz

jukuisma replied:
Done here: 214cc2269ef4e0b4a9e37f07d9a42353e07c1bd4.

halkoma commented on bin/findedit:
```diff
@@ -0,0 +1,20 @@
+fi
+
+searchterm="$1"
+file=$(
```
```suggestion
selection=$(
```

jukuisma replied:
Done here: 214cc2269ef4e0b4a9e37f07d9a42353e07c1bd4.

jukuisma committed:
Final findedit touch ups

1) Rename findedit -> fe for better ergonomics.
2) Fix bat preview.
3) Highlight the matched line.
